vicon/udp.h

```diff
@@ -282,16 +314,16 @@ class UDPClient
                     memcpy(&itemDataSize, &buffer[itemOffset + 1], sizeof(uint16_t));
                     assert(itemDataSize == 72);
 
-                    // Read object translation
-                    double translation[3];
-                    memcpy(&translation[0], &buffer[itemOffset + 27], 3 * sizeof(double));
+                    // Read object position
+                    Vector<millimeter_t> position;
+                    memcpy(&position[0], &buffer[itemOffset + 27], 3 * sizeof(double));
 
-                    // Read object rotation
-                    double rotation[3];
-                    memcpy(&rotation[0], &buffer[itemOffset + 51], 3 * sizeof(double));
+                    // Read object attitude
+                    Vector<radian_t> attitude;
+                    memcpy(&attitude[0], &buffer[itemOffset + 51], 3 * sizeof(double));
```

Comment:
The reliance on ``millimeter_t`` etc using double as it's underlying store AND ``sizeof(millimeter_t) == sizeof(double)`` worries me a little here - I feel these variables could be passed to ``updateObjectData`` as raw doubles (the units are always the same after all) and then, in ``ObjectData`` they can be converted for the user.

Reply:
This makes sense, especially as the Vicon system gives the rotation angles in a strange order.